src/aura/BGE_EntryForm/BGE_EntryFormController.js

```diff
@@ -104,20 +106,26 @@
                         header: component.get('v.donationModalHeader'),
                         body: newcomponent,
                         showCloseButton: true,
-                        cssClass: 'slds-modal_large'
+                        cssClass: 'slds-modal_large',
+                        closeCallback: function() {
+                            component.set('v.modalOpen', false);
+                        }
                     });
                 } else if (status === 'INCOMPLETE') {
```

Comment:
These two else if blocks could be somewhat combined within an outer "else".  The declaration of the message variable, the "component.set('v.modalOpen', false);" (or new setModalOpenToFalse()) and the call to helper.sendMessage could then be written only once within the outer "else" block, while the contents of the message can be generated within the inner if block that checks the error status...

Reply:
Hey Ken, what do you think of something like this inside the outer "else"?

setModalOpenToFalse();
const message = {
    title: $A.get('$Label.c.PageMessagesError'),
    errorMessage: status === 'ERROR' ? errorMessage : $A.get('$Label.c.stgUnknownError')
 };
helper.sendMessage('onError', message);

Reply:
Hey Prince, Yep, structured like this would be great!